Anlab.Mvc/Controllers/ReviewerController.cs

```diff
@@ -129,6 +130,92 @@ public async Task<IActionResult> EmailList(DateTime? start, DateTime? end, strin
             return View(model);
         }
 
+
+        public async Task<IActionResult> HistoricalSales(DateTime? start, DateTime? end)
+        {
+            var model = new HistoricalSalesModel();
+            if (start == null && end == null)
+            {
+                model.Start = new DateTime(DateTime.UtcNow.Year, 1, 1).Date;
+                model.End = DateTime.UtcNow.ToPacificTime().Date;
+            }
+            else
+            {
+                model.Start = start?.Date;
+                model.End = end?.Date;
+            }
+
+            var query = _context.HistoricalSalesViews.AsQueryable();
+            if (model.Start != null)
+            {
+                query = query.Where(a => a.DateFinalized >= model.Start.Value.Date.FromPacificTime());
+            }
+            if (model.End != null)
+            {
+                query = query.Where(a => a.DateFinalized <= model.End.Value.Date.AddDays(1).FromPacificTime());
+            }
+
+            var results = await query.ToListAsync(); //Grab all the results for the date range for processing

```

Comment:
If you use .ToView() it probably won't matter, but doing `AsNoTracking()` will save you some memory overhead.